src/main/java/com/emc/object/s3/S3Client.java

```diff
@@ -110,6 +110,16 @@ public interface S3Client {
      */
     void deleteBucket(String bucketName);
 
+    /**
+     * Deletes <code>bucketName</code>. The bucket could contain objects and versions before it can be deleted
+     */
+    void deleteBucket(DeleteBucketRequest request);
+
+    /**
+     * Gets status of background bucket cleanup tasks for <code>bucketName</code> if present
+     */
+    BucketDeletionStatus getBucketDeletionStatus(String bucketName);
```

Comment:
This API is tested for bucket with background deletion, what's the behavior for bucket not being deleted? Better add test case or more description.